### ISSUE TITLE
borderColor property doesn't work

### ISSUE THREAD

**Issue by tomoyuki-hiranuma** [kuma-ui/kuma-ui#355]
# Description
`borderColor` property doesn't work correctly when using with `color` property

### Code
```typescript
// example.tsx
const Sample = () => {
  return (
      <Box color='GrayText' border='1px solid' borderColor='orange'>
        test
      </Box>
  )
}
```

### Styles in developer tool
![image](https://github.com/kuma-ui/kuma-ui/assets/68529763/159e360b-3047-4a68-8895-72f6c86e1486)

### The actual UI
![image](https://github.com/kuma-ui/kuma-ui/assets/68529763/af6cf396-bc41-4e68-899c-e32428f5371b)

**Comment by yukukotani** [comment]
@tomoyuki-hiranuma Thank you for the reporting!

It seems that `border` overrides `border-color` in browsers. We can avoid this by putting `border-color` after `border`, but now Kuma doesn't respect the order of properties.

As a workaround, please avoid using grouped properties like `border` and split them like this.

```tsx
<Box color='GrayText' borderWidth='1px' borderStyle='solid' borderColor='orange'>
  test
</Box>
```

**Comment by poteboy** [comment]
Fixed in #367
